From 16ca33bafc4b2bab1712063e3e4057f01082d2a9 Mon Sep 17 00:00:00 2001
From: Du Chenyang <chenyang.du@intel.com>
Date: Tue, 28 May 2013 14:56:45 +0800
Subject: [PORT FROM MAIN] Add -m32/-melf_i386 for x86 target gcc/ld

BZ: 82732

This change simply makes the target arch explicit.
This is necessary for newer versions of gcc, when building on 64-bit hosts,
as the gcc will otherwise assume the target arch is the same as the host.

Change-Id: I6a4a7f2533a20493d13c003b99477ff5bbe1911e
Orig-Change-Id: Ifb938b8a2b3c875cf4bcdb04e69c0bc102a40c6b
Signed-off-by: Pavel Chupin <pavel.v.chupin@intel.com>
Reviewed-on: http://android.intel.com:8080/86279
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Smith, Geoff <geoff.smith@intel.com>
Reviewed-by: Yang, Fei <fei.yang@intel.com>
Reviewed-by: Yin, Fengwei <fengwei.yin@intel.com>
Tested-by: Ng, Cheon-woei <cheon-woei.ng@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libc/Android.mk |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/libc/Android.mk b/libc/Android.mk
index 6448d5d..0b07cd1 100644
--- a/libc/Android.mk
+++ b/libc/Android.mk
@@ -497,7 +497,8 @@ endif # !arm
 
 ifeq ($(TARGET_ARCH),x86)
   libc_common_cflags += -DSOFTFLOAT
-  libc_crt_target_cflags :=
+  libc_crt_target_cflags := -m32
+  libc_crt_target_ldflags := -melf_i386
   ifeq ($(ARCH_X86_HAVE_SSE2),true)
       libc_crt_target_cflags += -DUSE_SSE2=1
   endif
@@ -633,7 +634,7 @@ ALL_GENERATED_SOURCES += $(GEN)
 GEN := $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_static.o
 $(GEN): $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_static1.o $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbrand.o
 	@mkdir -p $(dir $@)
-	$(hide) $(TARGET_LD) -r -o $@ $^
+	$(hide) $(TARGET_LD) $(libc_crt_target_ldflags) -r -o $@ $^
 ALL_GENERATED_SOURCES += $(GEN)
 
 GEN := $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_dynamic1.o
@@ -648,7 +649,7 @@ ALL_GENERATED_SOURCES += $(GEN)
 GEN := $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_dynamic.o
 $(GEN): $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbegin_dynamic1.o $(TARGET_OUT_INTERMEDIATE_LIBRARIES)/crtbrand.o
 	@mkdir -p $(dir $@)
-	$(hide) $(TARGET_LD) -r -o $@ $^
+	$(hide) $(TARGET_LD) $(libc_crt_target_ldflags) -r -o $@ $^
 ALL_GENERATED_SOURCES += $(GEN)
 
 # We rename crtend.o to crtend_android.o to avoid a
-- 
1.7.4.1

